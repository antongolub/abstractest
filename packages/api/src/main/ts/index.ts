export * from './runner'
export * from './interface'
export * from './util'
