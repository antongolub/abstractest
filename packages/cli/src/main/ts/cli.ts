#!/usr/bin/env node

import {run} from '@abstractest/core'
import minimist from 'minimist'
import {createRequire} from 'node:module'

const camelize = (s: string) => s.replace(/-./g, x => x[1].toUpperCase())
const normalizeFlags = (flags = {}): Record<string, any> => Object.entries(flags).reduce((acc, [k, v]) => ({...acc, [camelize(k)]: v}), {})

const {_, cwd, runner, help, version, injectGlobal} = normalizeFlags(minimist(process.argv.slice(2), {
  alias: {
    help: ['h'],
    version: ['v'],
    'inject-global': ['inject-globals']
  },
  boolean: ['inject-global']
}));

(() => {

if (help) {
  console.log(`
  Usage:
    $ abstractest <pattern>
  
  Options:
    --runner        Specify test runner: jest, native. Defaults to void
    --help, -h      Print help digest
    --version, -v   Print version
  
  Examples:
    $ abstractest --runner=jest src/test/**/*.js
`)
  return
}

if (version) {
  console.log((import.meta.url ? createRequire(import.meta.url) : require)('../../package.json').version)
  return
}

run({
  cwd,
  include: _,
  runner,
  injectGlobal,
})

})()

