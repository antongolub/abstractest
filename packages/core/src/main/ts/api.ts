import {Describe, SuiteFn, It, TestFn, Expect, Hook} from './interface'
import {getRunner} from './runner'

export const it: It = Object.assign((name: string, fn?: TestFn) => getRunner().api.it(name, fn), {
  only(name: string, fn?: TestFn) { return getRunner().api.it.only(name, fn) },
  skip(name: string, fn?: TestFn) { return getRunner().api.it.skip(name, fn) },
  todo(name: string, fn?: TestFn) { return getRunner().api.it.todo(name, fn) },
})

export const describe: Describe = Object.assign((name: string, fn?: SuiteFn) => getRunner().api.describe(name, fn), {
  only(name: string, fn?: SuiteFn) { return getRunner().api.describe.only(name, fn) },
  skip(name: string, fn?: SuiteFn) { return getRunner().api.describe.skip(name, fn) },
  todo(name: string, fn?: SuiteFn) { return getRunner().api.describe.todo(name, fn) },
})

export const expect: Expect = (value) =>         getRunner().api.expect(value)

export const before: Hook = (fn) =>     getRunner().api.before(fn)

export const beforeEach: Hook = (fn) => getRunner().api.beforeEach(fn)

export const after: Hook = (fn) =>      getRunner().api.after(fn)

export const afterEach: Hook = (fn) =>  getRunner().api.afterEach(fn)
