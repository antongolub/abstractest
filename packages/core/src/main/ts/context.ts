const scope = '__ABSTRACTEST__'
const g: any = globalThis

export const context = {
  get data() {
    if (!g[scope]) {
      Object.defineProperty(g, scope, {
        enumerable: false,
        configurable: false,
        value: {
          global: {...g},
          runners: new Map()
        }
      })
    }
    return g[scope]
  },
  get runners() {
    return this.data.runners
  },
  get runner() {
    return this.data.runner
  },
  set runner(name: string) {
    this.data.runner = name
  }
}
