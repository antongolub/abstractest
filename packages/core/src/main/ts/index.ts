export * from './api'
export * from './context'
export * from './interface'
export * from './runner'
export * from './util'
