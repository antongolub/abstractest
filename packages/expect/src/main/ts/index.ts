import { expect as _expect } from 'expect'
import type {Expect} from '@abstractest/types'
import {
  toMatchSnapshot,
  toMatchInlineSnapshot,
  toThrowErrorMatchingInlineSnapshot,
  toThrowErrorMatchingSnapshot,
  addSerializer,
} from 'jest-snapshot'

// https://github.com/jestjs/jest/blob/main/packages/jest-expect/src/index.ts
_expect.extend({
  // https://github.com/facebook/jest/blob/e0b33b74b5afd738edc183858b5c34053cfc26dd/e2e/custom-inline-snapshot-matchers/__tests__/bail.test.js
  toMatchSnapshot(this: any,...args: any) {
    this.dontThrow = () => {}
    return toMatchSnapshot.apply(this, args)
  },
  toMatchInlineSnapshot(this: any,...args: any) {
    this.dontThrow = () => {}
    return toMatchInlineSnapshot.apply(this, args)
  },
  // @ts-ignore
  toThrowErrorMatchingInlineSnapshot,
  // @ts-ignore
  toThrowErrorMatchingSnapshot,
})

// @ts-ignore
_expect.addSnapshotSerializer = addSerializer

export const expect: Expect = _expect as any

export {SnapshotState} from 'jest-snapshot'
