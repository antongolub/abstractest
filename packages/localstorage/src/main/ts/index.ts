import type {Mocker} from '@abstractest/types'
import { LocalStorage } from './localstorage'
export { LocalStorage } from './localstorage'

const w: any = globalThis || window || (global && global.window) || global

export const init = (mocker?: Mocker) => {
  const mock = mocker || w.jest
  w.localStorage = new LocalStorage(mock)
  w.sessionStorage = new LocalStorage(mock)
}

export const destroy = () => {
  delete w.sessionStorage
  delete w.localStorage
}
