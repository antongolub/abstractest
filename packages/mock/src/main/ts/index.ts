import {ModuleMocker} from 'jest-mock'

export const mock = new ModuleMocker(globalThis)
