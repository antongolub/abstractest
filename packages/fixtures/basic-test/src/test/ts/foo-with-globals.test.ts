import {foo} from '../../main/ts/foo'
import '@abstractest/types'

describe('with global test API', () => {
  it('foo() returns bar (async)', () => {
    expect(foo()).toEqual('bar')
  })
})
