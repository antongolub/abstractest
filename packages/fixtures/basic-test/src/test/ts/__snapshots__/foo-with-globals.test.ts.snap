// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with global test API foo() returns bar (async) 1`] = `"bar"`;
